spec/models/project_spec.rb

```diff
@@ -4,296 +4,1071 @@
+
+        context "when project state requires a video url" do
+          context "state is online" do
+            context "and goal is equals minimum" do
+              let(:project) { create(:project, :online, goal: 5000) }
+
+              it { is_expected.not_to allow_value(nil).for(:video_url) }
+            end
+
+            context "and goal is above minimum" do
+              let(:project) { create(:project, :online, goal: 5001) }
+
+              it { is_expected.not_to allow_value(nil).for(:video_url) }
+            end
+
+            context "and goal is below minimum" do
+              let(:project) { create(:project, :online, goal: 4999) }
+
+              it { is_expected.to allow_value(nil).for(:video_url) }
+            end
+          end
+
+          context "state is in analysis" do
+            context "and goal is equals minimum" do
+              let(:project) { create(:project, :in_analysis, goal: 5000) }
+
+              it { is_expected.not_to allow_value(nil).for(:video_url) }
+            end
+
+            context "and goal is above minimum" do
+              let(:project) { create(:project, :in_analysis, goal: 5001) }
+
+              it { is_expected.not_to allow_value(nil).for(:video_url) }
+            end
+
+            context "and goal is below minimum" do
+              let(:project) { create(:project, :in_analysis, goal: 4999) }
+
+              it { is_expected.to allow_value(nil).for(:video_url) }
+            end
+          end
+        end
+
+        context "when project state does not require a video url" do
+          context "state is waiting_funds" do
+            let(:project) { create(:project, :waiting_funds) }
+
+            it { is_expected.to allow_value(nil).for(:video_url) }
+          end
+
+          context "state is successful" do
+            let(:project) { create(:project, :successful) }
+
+            it { is_expected.to allow_value(nil).for(:video_url) }
+          end
+        end
+      end
+    end
   end
 
-  describe ".of_current_week" do
-    subject { Project.of_current_week }
-    before do
-      3.times { create(:project, state: 'online', online_date: DateTime.now) }
-      3.times { create(:project, state: 'draft', online_date: 3.days.ago) }
-      3.times { create(:project, state: 'successful', online_date: 6.days.ago) }
-      5.times { create(:project, state: 'online', online_date: 8.days.ago) }
-      5.times { create(:project, state: 'online', online_date: 2.weeks.ago) }
-      2.times { create(:project, state: 'in_analysis', online_date: 3.days.from_now)}
+  describe ".search_on_name" do
+    let(:category) { create(:category, name_pt: 'teste_pt', name_en: 'test_en') }
+    let!(:project) { create(:project, name: 'foo', category: category) }
+
+    context "when project is found" do
+      context "and it searches on category name" do
+        context "and it's in portuguese" do
+          subject { Project.search_on_name('teste_pt').map(&:name) }
```

Comment:
Hum....

That is happening because we you are using the shortcut `it {}`, If I were you would write explicity.
`it "bla bla"`.

But let's see what the others think about this.